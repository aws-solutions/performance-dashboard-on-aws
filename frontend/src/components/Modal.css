.modal {
  position: absolute;
  right: 40px;
  bottom: 40px;
  border-left: 1px solid #ccc;
  border-top: 1px solid #ccc;
  border-right: 2px solid #333;
  border-bottom: 2px solid #333;
  background: #fff;
  overflow: auto;
  outline: none;
  padding: 20px;
  top: 50%;
  left: 50%;
  width: 512px;
  display: table;
  transform: translate(-50%, -50%);
  -webkit-overflow-scrolling: touch;
  z-index: 100001;
}

.overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.75);
  z-index: 100000;
}
