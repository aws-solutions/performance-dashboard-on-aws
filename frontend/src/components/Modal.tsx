/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faTimes } from "@fortawesome/free-solid-svg-icons";
import Button from "../components/Button";
import ReactModal from "react-modal";
import "./Modal.css";
import { useTranslation } from "react-i18next";

interface PathParams {
    isOpen: boolean;
    closeModal: () => void;
    title: string;
    message: string | React.ReactNode;
    buttonType: string;
    buttonAction: () => void;
    ariaHideApp?: boolean;
}

function Modal(props: PathParams) {
    const { t } = useTranslation();
    return (
        <ReactModal
            isOpen={props.isOpen}
            onRequestClose={() => {
                props.closeModal();
            }}
            className="modal usa-modal"
            overlayClassName="overlay"
            shouldFocusAfterRender={true}
            shouldReturnFocusAfterClose={true}
            aria={{
                labelledby: t("title"),
                describedby: t("message"),
                modal: "true",
            }}
            ariaHideApp={true}
        >
            <div className="clearfix">
                <div
                    className="float-left"
                    style={{
                        maxWidth: "80%",
                    }}
                >
                    <h2 id="title" className="margin-top-0">
                        {props.title}
                    </h2>
                </div>
                <div className="float-right font-sans-md">
                    <Button
                        variant="unstyled"
                        type="button"
                        className="margin-left-1 text-base-dark hover:text-base-darker active:text-base-darkest"
                        onClick={props.closeModal}
                        ariaLabel={t("GlobalClose")}
                    >
                        <FontAwesomeIcon
                            icon={faTimes}
                            className="margin-right-1"
                            size="1x"
                            style={{ marginTop: "5px" }}
                        />
                    </Button>
                </div>
            </div>
            <div id="message" className="font-sans-md margin-top-2 margin-bottom-6">
                {props.message}
            </div>

            <Button type="button" variant="base" onClick={props.buttonAction}>
                {props.buttonType}
            </Button>
        </ReactModal>
    );
}

export default Modal;
