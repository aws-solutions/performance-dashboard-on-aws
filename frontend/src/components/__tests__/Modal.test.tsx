/*
 *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *  SPDX-License-Identifier: Apache-2.0
 */

import React from "react";
import { render } from "@testing-library/react";
import Modal from "../Modal";

test("renders a delete type Modal", async () => {
    const { baseElement } = render(
        <Modal
            title={"test title"}
            message={"test message"}
            isOpen={true}
            closeModal={() => {}}
            buttonType="Delete"
            buttonAction={() => {}}
            ariaHideApp={false}
        />,
    );
    expect(baseElement).toMatchSnapshot();
});

test("renders a publish type Modal", async () => {
    const { baseElement } = render(
        <Modal
            title={"test title"}
            message={"test message"}
            isOpen={true}
            closeModal={() => {}}
            buttonType="Prepare for publishing"
            buttonAction={() => {}}
            ariaHideApp={false}
        />,
    );
    expect(baseElement).toMatchSnapshot();
});

test("renders an archive type Modal", async () => {
    const { baseElement } = render(
        <Modal
            title={"test title"}
            message={"test message"}
            isOpen={true}
            closeModal={() => {}}
            buttonType="Archive"
            buttonAction={() => {}}
            ariaHideApp={false}
        />,
    );
    expect(baseElement).toMatchSnapshot();
});

test("renders a create draft type Modal", async () => {
    const { baseElement } = render(
        <Modal
            title={"test title"}
            message={"test message"}
            isOpen={true}
            closeModal={() => {}}
            buttonType="Create draft"
            buttonAction={() => {}}
            ariaHideApp={false}
        />,
    );
    expect(baseElement).toMatchSnapshot();
});

test("renders a re-publish type Modal", async () => {
    const { baseElement } = render(
        <Modal
            title={"test title"}
            message={"test message"}
            isOpen={true}
            closeModal={() => {}}
            buttonType="Re-publish"
            buttonAction={() => {}}
            ariaHideApp={false}
        />,
    );
    expect(baseElement).toMatchSnapshot();
});
