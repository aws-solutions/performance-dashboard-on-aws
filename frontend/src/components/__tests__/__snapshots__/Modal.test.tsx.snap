// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a create draft type Modal 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay overlay"
    >
      <div
        aria-describedby="message"
        aria-labelledby="title"
        aria-modal="true"
        class="ReactModal__Content modal usa-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="clearfix"
        >
          <div
            class="float-left"
            style="max-width: 80%;"
          >
            <h2
              class="margin-top-0"
              id="title"
            >
              test title
            </h2>
          </div>
          <div
            class="float-right font-sans-md"
          >
            <button
              aria-label="Close"
              class="usa-button usa-button--unstyled margin-left-1 text-base-dark hover:text-base-darker active:text-base-darkest"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times fa-w-11 fa-1x margin-right-1"
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="margin-top: 5px;"
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="font-sans-md margin-top-2 margin-bottom-6"
          id="message"
        >
          test message
        </div>
        <button
          class="usa-button usa-button--base"
          type="button"
        >
          Create draft
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders a delete type Modal 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay overlay"
    >
      <div
        aria-describedby="message"
        aria-labelledby="title"
        aria-modal="true"
        class="ReactModal__Content modal usa-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="clearfix"
        >
          <div
            class="float-left"
            style="max-width: 80%;"
          >
            <h2
              class="margin-top-0"
              id="title"
            >
              test title
            </h2>
          </div>
          <div
            class="float-right font-sans-md"
          >
            <button
              aria-label="Close"
              class="usa-button usa-button--unstyled margin-left-1 text-base-dark hover:text-base-darker active:text-base-darkest"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times fa-w-11 fa-1x margin-right-1"
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="margin-top: 5px;"
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="font-sans-md margin-top-2 margin-bottom-6"
          id="message"
        >
          test message
        </div>
        <button
          class="usa-button usa-button--base"
          type="button"
        >
          Delete
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders a publish type Modal 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay overlay"
    >
      <div
        aria-describedby="message"
        aria-labelledby="title"
        aria-modal="true"
        class="ReactModal__Content modal usa-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="clearfix"
        >
          <div
            class="float-left"
            style="max-width: 80%;"
          >
            <h2
              class="margin-top-0"
              id="title"
            >
              test title
            </h2>
          </div>
          <div
            class="float-right font-sans-md"
          >
            <button
              aria-label="Close"
              class="usa-button usa-button--unstyled margin-left-1 text-base-dark hover:text-base-darker active:text-base-darkest"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times fa-w-11 fa-1x margin-right-1"
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="margin-top: 5px;"
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="font-sans-md margin-top-2 margin-bottom-6"
          id="message"
        >
          test message
        </div>
        <button
          class="usa-button usa-button--base"
          type="button"
        >
          Prepare for publishing
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders a re-publish type Modal 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay overlay"
    >
      <div
        aria-describedby="message"
        aria-labelledby="title"
        aria-modal="true"
        class="ReactModal__Content modal usa-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="clearfix"
        >
          <div
            class="float-left"
            style="max-width: 80%;"
          >
            <h2
              class="margin-top-0"
              id="title"
            >
              test title
            </h2>
          </div>
          <div
            class="float-right font-sans-md"
          >
            <button
              aria-label="Close"
              class="usa-button usa-button--unstyled margin-left-1 text-base-dark hover:text-base-darker active:text-base-darkest"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times fa-w-11 fa-1x margin-right-1"
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="margin-top: 5px;"
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="font-sans-md margin-top-2 margin-bottom-6"
          id="message"
        >
          test message
        </div>
        <button
          class="usa-button usa-button--base"
          type="button"
        >
          Re-publish
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`renders an archive type Modal 1`] = `
<body
  class="ReactModal__Body--open"
>
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div />
  <div
    data-react-modal-body-trap=""
    style="position: absolute; opacity: 0;"
    tabindex="0"
  />
  <div
    class="ReactModalPortal"
  >
    <div
      class="ReactModal__Overlay overlay"
    >
      <div
        aria-describedby="message"
        aria-labelledby="title"
        aria-modal="true"
        class="ReactModal__Content modal usa-modal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="clearfix"
        >
          <div
            class="float-left"
            style="max-width: 80%;"
          >
            <h2
              class="margin-top-0"
              id="title"
            >
              test title
            </h2>
          </div>
          <div
            class="float-right font-sans-md"
          >
            <button
              aria-label="Close"
              class="usa-button usa-button--unstyled margin-left-1 text-base-dark hover:text-base-darker active:text-base-darkest"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-times fa-w-11 fa-1x margin-right-1"
                data-icon="times"
                data-prefix="fas"
                focusable="false"
                role="img"
                style="margin-top: 5px;"
                viewBox="0 0 352 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
                  fill="currentColor"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="font-sans-md margin-top-2 margin-bottom-6"
          id="message"
        >
          test message
        </div>
        <button
          class="usa-button usa-button--base"
          type="button"
        >
          Archive
        </button>
      </div>
    </div>
  </div>
</body>
`;
